src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1800,20 +1812,32 @@ public ImmutableArray<DocumentId> GetDocumentIdsWithFilePath(string? filePath)
         private ImmutableDictionary<ProjectId, CompilationTracker> CreateCompilationTrackerMap(ProjectId projectId, ProjectDependencyGraph dependencyGraph)
         {
             var builder = ImmutableDictionary.CreateBuilder<ProjectId, CompilationTracker>();
-            var dependencies = dependencyGraph.GetProjectsThatTransitivelyDependOnThisProject(projectId);
+            IEnumerable<ProjectId>? dependencies = null;
 
-            foreach (var projectIdAndTracker in _projectIdToTrackerMap)
+            foreach (var (id, tracker) in _projectIdToTrackerMap)
             {
-                var id = projectIdAndTracker.Key;
-                var tracker = projectIdAndTracker.Value;
-
                 if (!tracker.HasCompilation)
                 {
                     continue;
                 }
 
-                var canReuse = id == projectId || !dependencies.Contains(id);
-                builder.Add(id, canReuse ? tracker : tracker.Fork(tracker.ProjectState));
+                builder.Add(id, CanReuse() ? tracker : tracker.Fork(tracker.ProjectState));
+
+                // Returns true if 'tracker' can be reused for project 'id'
+                bool CanReuse()
+                {
+                    if (id == projectId)
+                        return true;
+
+                    var forwardDependencies = dependencyGraph.TryGetProjectsThatThisProjectTransitivelyDependsOn(id);
+                    if (forwardDependencies is object && !forwardDependencies.Contains(projectId))
+                    {
+                        return true;
+                    }
```

Comment:
can you make the braces consistent with the check above?

Reply:
Current bracing adheres to policy for this project.

Reply:
can you still make it consistent with the braces above.  it's not about adhering to policy or not.  it's about being needlessly inconsistent with new code that is being written. i.e. tons of code could be written that would follow policy but still be a mishmash of inconsistent styling *with itself*.  That's not a good thing.